cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -770,6 +774,30 @@ public void deleteArtifact(Id.Artifact artifactId) throws Exception {
     privilegesManager.revoke(artifactId.toEntityId());
   }
 
+  /**
+   * return list of {@link ArtifactInfo} in the namespace
+   * @param namespace
+   * @return list of {@link ArtifactInfo}
+   * @throws Exception
+   */
+  public List<ArtifactInfo> getArtifactsInfo(NamespaceId namespace) throws Exception {
+    final List<ArtifactDetail> artifactDetails = artifactStore.getArtifacts(namespace);
+
+    List<ArtifactInfo> artifactInfoList =
+      Lists.transform(artifactDetails, new Function<ArtifactDetail, ArtifactInfo>() {
+        @Nullable
+        @Override
+        public ArtifactInfo apply(@Nullable ArtifactDetail input) {
+          // transform artifactDetail to artifactInfo
+          ArtifactId artifactId = input.getDescriptor().getArtifactId();
+          return new ArtifactInfo(artifactId.getName(), artifactId.getVersion().getVersion(), artifactId.getScope(),
+                                  input.getMeta().getClasses(), input.getMeta().getProperties(),
+                                  input.getMeta().getUsableBy());
+        }
+      });
+    return Collections.unmodifiableList(filterAuthorizedArtifactInfos(artifactInfoList, namespace));
```

Comment:
is it okay to do that in a follow-up PR ? it involves couple of changes and test cases, don't want to make this PR longer.